crates/credentials/src/presentation_definition_v2.rs

```diff
@@ -0,0 +1,230 @@
+use jsonschema::{Draft, JSONSchema};
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+use serde_with::skip_serializing_none;
+use std::collections::HashMap;
+
+/// Presentation Exchange
+///
+/// Presentation Exchange specification codifies a Presentation Definition data format Verifiers
+/// can use to articulate proof requirements, and a Presentation Submission data format Holders can
+/// use to describe proofs submitted in accordance with them.
+///
+/// See [Presentation Definition](https://identity.foundation/presentation-exchange/#presentation-definition)
+/// for more information.
+#[skip_serializing_none]
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
```

Comment:
Curious why the `Debug, Default, PartialEq` are needed?

Reply:
`derive` is generating additional code at compile time for this struct:
* `Debug` so I can print out the values and get a debug description of the struct & all of its fields
* `Default` so I can create this struct with pre-populated default values (used in tests)
* `PartialEq` so I can use the `==` operator in tests